docs/Testing.md

```diff
@@ -136,79 +46,106 @@ try {
   // handle error
 }
 ```
-Example:
-```js
-// server.js
-const minimist = require('minimist')
-const fastify = require('fastify')()
-
-const options = {
-  schema: {
-    response: {
-      200: {
-        type: 'object',
-        properties: {
-          hello: { type: 'string' }
-        }
-      }
-    }
-  }
-}
 
-fastify.get('/', options, function (request, reply) {
-  reply.send({ hello: 'world' })
-})
+#### Example:
 
-function start (opts, callback) {
-  fastify.listen(opts.port, function (err) {
-    callback(err, fastify)
-  })
-}
+**app.js**
+```js
+const Fastify = require('fastify')
 
-// In this way you can run the server both from the CLI and as a required module.
-if (require.main === module) {
-  // Run the server with:
-  // $ node server.js -p 8080
-  start(minimist(process.argv.slice(2), {
-    integer: ['port'],
-    alias: {
-      port: 'p'
-    },
-    default: {
-      port: 3000
-    }
-  }), (err, instance) => {
-    if (err) throw err
-
-    console.log(`server listening on ${instance.server.address().port}`)
+function buildFastify () {
+  const fastify = Fastify()
+
+  fastify.get('/', function (request, reply) {
+    reply.send({ hello: 'world' })
   })
+  
+  return fastify
 }
 
-// note that now we are also exposing the fastify instance
-module.exports = { start, fastify }
+module.exports = buildFastify
 ```
 
+**test.js**
 ```js
-// test.js
-const t = require('tap')
-const test = t.test
-const fastify = require('./server').fastify
-
-test('GET `/` route', t => {
-  t.plan(3)
+const tap = require('tap')
+const buildFastify = require('./app')
+
+tap.test('GET `/` route', t => {
+  t.plan(4)
+  
+  const fastify = buildFastify()
+  
+  // Even if the server is not running (inject does not run the server),
```

Comment:
this is false. `inject` have fastify run the full "boot" cycle, so it is _running_, in some lateral sense.